crates/topos-sequencer-subnet-runtime/src/proxy.rs

```diff
@@ -352,6 +363,34 @@ impl SubnetRuntimeProxy {
         }
     }
 
+    async fn process_block(
+        subnet_runtime_proxy: Arc<Mutex<SubnetRuntimeProxy>>,
+        certification: Arc<Mutex<Certification>>,
+        block_info: BlockInfo,
+    ) -> Result<(), Error> {
+        let mut certification = certification.lock().await;
+        let block_number = block_info.number;
+
+        // Update certificate block history
+        certification.append_blocks(vec![block_info]);
+
+        let new_certificates = match certification.generate_certificates().await {
+            Ok(certificates) => certificates,
+            Err(e) => {
+                error!("Unable to generate certificates: {e}");
+                return Err(e);
+            }
+        };
+
+        debug!("Generated new certificates {new_certificates:?}");
+
+        for cert in new_certificates {
+            Self::send_new_certificate(subnet_runtime_proxy.clone(), cert, block_number).await
+        }
```

Comment:
Dumb q: what happens if we don't `await` here? We don't expect a return result here and `send_new_certificate` cannot fail (right?), so perhaps "fire and forget" could work?

Reply:
It will block here only if mpsc channel for passing certificates is full, otherwise it will resolve immediately.

Other general point: I am not if favor of  fire and forget approach in sequencer design. We should not create new certificates if pipeline for processing certificates is broken, better stop earlier. Anyway creating certificates depend on new block (which happens every 2 seconds), so not being able to pass new certificate in that time-frame is really some erroneous situation.

Reply:
>  I am not if favor of fire and forget approach in sequencer design.

I'm not a big fan either, but isn't that exactly what we're doing here? We send off the cert and then `await`, but we don't look at the outcome. If `send_new_certificates` resolves, but with an `Err`, we wouldn't know right?

Whenever we can, I think we should avoid relying on unwritten assumptions like for instance the 2 sec blocktime.

Reply:
He we log the error earlier if channel fails and return. We can discuss if we should panic or something else

Reply:
I will update this mechanism to return error instead of logging it

Reply:
@dvdplm I'll handle update of certificate generation and dispatching in a separate PR, as there are more things to optimize and cleanup